neuraxle/pipeline.py

```diff
@@ -70,8 +60,23 @@ class Pipeline(BasePipeline, ResumableStepMixin):
     Fits and transform steps after latest checkpoint
     """
 
-    def __init__(self, steps: NamedTupleList):
-        super().__init__(steps)
+    def __init__(self, steps: NamedTupleList, hasher=HasherByIndex()):
+        if hasher is None:
+            raise ValueError('Pipeline hasher cannot be None')
```

Comment:
todo: = None and add an if that creates it in the constructor. We want to avoid default arguments by reference. Same commen as in BaseStep's constructor.